solr/core/src/java/org/apache/solr/cloud/Overseer.java

```diff
@@ -857,13 +857,18 @@ private void doCompatCheck(BiConsumer<String, Object> consumer) {
       return;
     }
 
-    try (CloudSolrClient client =
-        new CloudHttp2SolrClient.Builder(
-                Collections.singletonList(getZkController().getZkServerAddress()), Optional.empty())
-            .withZkClientTimeout(30000, TimeUnit.MILLISECONDS)
-            .withZkConnectTimeout(15000, TimeUnit.MILLISECONDS)
-            .withHttpClient(getCoreContainer().getDefaultHttpSolrClient())
-            .build()) {
+    try (var solrClient =
```

Comment:
since this client is embedded into the Cloud one, I think you should name this var maybe internalClient or something and don't refer to it in the block of code for the try-finally except for passing into the CloudSolrClient.

BTW; it really is painful to customize these particular timeouts as we're forced to do it at an inner layer.  Ugh.  @epugh not sure if you worked on some of the SolrClient building methods and saw this issue before.  Maybe we should add the same methods to the CloudHttp2SolrClient.Builder